cli/internal/globby/globby.go

```diff
@@ -93,18 +102,13 @@ func globFilesFs(fs afero.IOFS, basePath string, includePatterns []string, exclu
 	includePattern = filepath.ToSlash(includePattern)
 	excludePattern = filepath.ToSlash(excludePattern)
 
-	err := doublestar.GlobWalk(fs, includePattern, func(path string, dirEntry iofs.DirEntry) error {
-		// Unix root paths do not prepend the leading slash.
-		if basePath == "/" && !strings.HasPrefix(path, "/") {
-			path = filepath.Join(basePath, path)
-		}
-
+	err := doublestar.GlobWalk(fsys, includePattern, func(path string, dirEntry iofs.DirEntry) error {
 		if dirEntry.IsDir() {
 			return nil
 		}
 
 		if excludeCount == 0 {
-			result.Add(path)
+			result.Add(root + path)
```

Comment:
I had to double-check this line, and the equivalent below. Maybe we need a note that `root` includes a trailing separator, and `path` does not include a leading separator? If we ever move to some globbing code that doesn't depend on `fs.FS`, I could see missing that as a mistake we might make.

Naïvely I would've expected this to be `filepath.Join`. Maybe it should be anyways?

Reply:
So, in an earlier iteration, I had the same hypothesis that`Join` with an empty string would do the trick. [But no, Go strikes again!](https://go.dev/play/p/Koh_hr_IsHG)

Really though, since `Join` hits the entirety of `Clean` underneath the hood, we should keep it as string concatenation. I'll do a better job explaining this.